Add inline module settings for plan welcome page

gravityrail opened the pull request:
Fixes #5909

#### Changes proposed in this Pull Request:

* Show toggles to enable plan features in the "welcome" page for the plan, after upgrade (as per #8700)

#### Testing instructions:

* Upgrade to Jetpack Personal
* Observe that form includes module toggles (toggle all on, then all off)
* Upgrade to Jetpack Premium
* Same as above
* Upgrade to Jetpack Professional
* Same as above

#### Screenshots

I have deliberately done the smallest change I can, while still sharing code across plans (which changes some copy). I have only removed prompts from the bottom of the form where I feel that there was a clearly redundant CTA.

### Personal

![goldsounds ngrok io_wp-admin_admin php_page jetpack ipad pro 3](https://user-images.githubusercontent.com/51896/37115465-0e3ac8c4-2200-11e8-9bce-134760d203f1.png)

### Premium

![goldsounds ngrok io_wp-admin_admin php_page jetpack ipad pro 2](https://user-images.githubusercontent.com/51896/37115477-158af464-2200-11e8-94dd-c486e16184f7.png)

### Professional

![goldsounds ngrok io_wp-admin_admin php_page jetpack ipad pro 1](https://user-images.githubusercontent.com/51896/37115504-21fe8378-2200-11e8-8369-0b6cca4b2df9.png)

alisterscott commented:
Hi, 

I tested this against jurassic.ninja sites (before and after the change) and it looked and worked well.

The only issue I came across was trying to activate ads on a personal site - it gave me an error:

<img width="1949" alt="personal" src="https://user-images.githubusercontent.com/128826/37126204-b92a2ad4-22bb-11e8-988a-605acdc9226f.png">

I had a look at both [the pricing page](https://jetpack.com/pricing/), and the plans page (in Calypso) and couldn't find monetization as part of the personal plan so maybe this doesn't belong in this change?

<img width="1066" alt="plans" src="https://user-images.githubusercontent.com/128826/37126201-b1360ef6-22bb-11e8-9c5e-a994e98216fe.png">

Otherwise everything else worked well for me for the other plans

ebinnion commented on _inc/client/components/welcome-new-plan/monitor-akismet-backups-prompt.jsx:
```diff
@@ -0,0 +1,30 @@
+		return <div>
+			<img src={ imagePath + 'security.svg' } className="jp-welcome__svg" alt={ __( 'Security' ) } />
+			<p>
+				{ __( 'Jetpack Professional gives you everything you need to keep your hard work safe, including ' +
```
This is shown for all three plans. Perhaps we could say, "Your Jetpack plan...". Or, we can add a placeholder in and then substitute the plan name in.

ebinnion commented on _inc/client/components/welcome-new-plan/wordads-prompt.jsx:
```diff
@@ -0,0 +1,28 @@
+/**
```
Are we using this component anywhere? I didn't notice it in the screenshots and when I search `wordads-prompt` on this page, nothing shows up.

ebinnion commented on _inc/client/components/welcome-new-plan/videopress-prompt.jsx:
```diff
@@ -0,0 +1,27 @@
+class VideoPressPrompt extends React.Component {
+	render() {
+		return <div>
+			<img src={ imagePath + 'wordads.svg' } className="jp-welcome__svg" alt={ __( 'VideoPress' ) } />
```
Since this icon is the same one we're using for the social and ads section, does it make sense to group the settings together under a single instance of the icon?

alisterscott commented:
Re-tested and I'm no longer seeing the ads on a personal plan - thanks for the fix.

<img width="747" alt="screen shot 2018-03-09 at 8 25 28 am" src="https://user-images.githubusercontent.com/128826/37180125-a4ef4d52-2373-11e8-860e-26ed6d21b086.png">

As @ebinnion points out above, could the `"Jetpack Professional gives you everything you need to keep your hard work safe,"` be rephrased as `"Your Jetpack plan gives you everything you need to keep your hard work safe,"`?

gravityrail commented:
@ebinnion I removed the unused modules, moved modules only used in one place to be inline, fixed the wording about "Professional" plan to be more generic, and removed the icon from the VideoPress section (since it didn't quite fit in the other section, and there was no videopress-suitable svg icon already in the repo)

oskosk commented:
@gravityrail this one needs a rebase now

gravityrail commented:
Merged with master and added toggle for enabling Search. Here's a screenshot - I put the search toggle right below the first mention of Search in the landing page. This might not be the most elegant way to show it (or perhaps we need an icon above the para/toggle?)

![goldsounds ngrok io_wp-admin_admin php_page jetpack ipad pro 4](https://user-images.githubusercontent.com/51896/37914139-5f281596-30cb-11e8-8fba-8404214f332d.png)

cc @Viper007Bond @keoshi

keoshi commented:
Looks great to me for a first iteration. Let's ship this!

oskosk commented:
@gravityrail I started with a site from scratch, connected, then upgraded to personal... Was presented with the toggles. three of them. Two of them were toggled. Only the backups one was off. So i toggled it, and then it activated but the welcome screen closed itself unexpectedly. Is this ok ?

oskosk commented:
Forget it. I can't reproduce. Tested all plans and looks great.

oskosk commented:
Note that the changes here, will not take advantage of the string updates done to the Learn More popover introduced in #9137.  cc @eliorivero @rickybanister

oskosk commented:
@rickybanister @eliorivero @keoshi I'm merging this one. Please let me know if we need to tackle the strings I mentioned on my previous comment

eliorivero commented:
Yeah, I think for consistency this should be implemented.
